hoomd/operation.py

```diff
@@ -144,153 +95,15 @@ def _setattr_typeparam(self, attr, value):
-            try:
-                # try to grab the namespace
-                state = deepcopy(NamespaceDict(data)[namespace])
-            except KeyError:
-                # if namespace can't be found assume that dictionary is the
-                # state dictionary (This assumes that values are of the form
-                # (value, flag)
-                try:
-                    state = dict_map(data, lambda x: x[0])
-                except TypeError:
-                    # if the map fails, we then assume that the dictionary is
-                    # one without the flag information on the data. This could
-                    # be the case if a logger backend stores the data and that
-                    # returned data is fed in.
-                    state = deepcopy(data)
-
-        # Data is of an unusable type
-        else:
-            raise ValueError("Object {} cannot be used to get state."
-                             "".format(data))
-
-        return (state, kwargs)
-
-    @classmethod
-    def _state_from_gsd(cls, filename, namespace, **kwargs):
-        """Get state dictionary from GSD file."""
-        if 'frame' not in kwargs.keys():
-            frame = -1
-        else:
-            frame = kwargs.pop('frame')
-        # Grab state keys from gsd
-        reader = GSDStateReader(filename, frame)
-        namespace_str = 'log/' + '/'.join(namespace)
-        state_chunks = reader.getAvailableChunks(namespace_str)
-        state_dict = NamespaceDict()
-        chunk_slice = slice(len(namespace_str) + 1, None)
-        # Build up state dict
-        for state_chunk in state_chunks:
-            state_dict_key = tuple(state_chunk[chunk_slice].split('/'))
-            state_dict[state_dict_key] = \
-                _handle_gsd_arrays(reader.readChunk(state_chunk))
-        return (state_dict._dict, kwargs)
-
-    @classmethod
-    def _from_state_with_state_dict(cls, state, **kwargs):
-        """Using the state dictionary create a new object."""
-        # Initialize object using params from state and passed arguments
-        params = state.get('__params__', {})
-        params.update(kwargs)
-        obj = cls(**params)
-        state.pop('__params__', None)
-
-        # Add typeparameter information
-        for name, tp_dict in state.items():
-            if '__default__' in tp_dict.keys():
-                obj._typeparam_dict[name].default = tp_dict['__default__']
-                del tp_dict['__default__']
-            # Parse the stringified tuple back into tuple
-            if obj._typeparam_dict[name]._len_keys > 1:
-                tp_dict = str_to_tuple_keys(tp_dict)
-            setattr(obj, name, tp_dict)
-        return obj
+    def __eq__(self, other):
+        if not isinstance(other, _HOOMDGetSetAttrBase):
+            return NotImplemented
+        return (self._param_dict == other._param_dict
+                and all(
+                    self._typeparam_dict.keys() == other._typeparam_dict.keys()
```

Comment:
I'm not sure if this needs to be a `set` -- are dictionary key views treated as ordered or unordered? Also probably depends on the implementation of `keys()` for this object.

Reply:
A set would be more reliable.